src/Encoder.php

```diff
@@ -0,0 +1,347 @@
+<?php
+
+namespace CodeRevolutionPlugins\GPT3Encoder;
+
+class Encoder
+{
+    private static ?self $instance = null;
+
+    public static function instance(): self
+    {
+        if (null === self::$instance) {
+            self::$instance = self::create();
+        }
+
+        return self::$instance;
+    }
+
+    private array $encoder;
+    private array $byte_encoder;
+    private array $bpe_ranks;
+    public function __construct($encoder, $byte_encoder, $bpe_ranks)
```

Comment:
```suggestion
    public function __construct(private array $encoder, private array $byte_encoder, private array $bpe_ranks)
```

depends on which PHP version you go with